cmd/compose/up.go

```diff
@@ -148,6 +148,7 @@ func upCommand(p *ProjectOptions, dockerCli command.Cli, backend api.Service) *c
 	flags.BoolVar(&create.noBuild, "no-build", false, "Don't build an image, even if it's policy")
 	flags.StringVar(&create.Pull, "pull", "policy", `Pull image before running ("always"|"missing"|"never")`)
 	flags.BoolVar(&create.removeOrphans, "remove-orphans", false, "Remove containers for services not defined in the Compose file")
+	flags.BoolVar(&create.recreateVolumes, "recreate-volumes", false, "Recreate volumes when volume configuration in the Compose file changes.")
```

Comment:
we can rename it :P

Reply:
I also wonder this could be ambiguous, considering we have `--renew-anon-volumes`. Need to noun that make it clear we recreate volumes _when_ config has diverged

Reply:
how about `update-volumes`?

Reply:
@ndeloof the `recreate-volumes` makes somewhat sense to me, also given the 'pattern' for `force-recreate` flag. I don't think it is ambiguous with the flag `--renew-anon-volumes`

Reply:
it is, as we only recreate diverged volumes. Not all volumes (and not anonymous ones)

Reply:
@ndeloof I can add that at the flag description: that this will ignore anonymous volumes. As for the names maybe? If you have any other ideas please feel free to suggest

--recreate-volumes-if-changed
--force-recreate-volumes
--sync-volumes
--update-volumes
--refresh-named-volumes
--rebuild-volumes

Reply:
Or do you prefer to make this the default behaviour with a prompt?